### ISSUE TITLE
The downward metrics server allows using more resources than expected

### ISSUE THREAD

**Issue by germag** [kubevirt/kubevirt#11731]
**What happened**:
The downward metric server could be abused to use an excessive amount of resources by the virt-handler if many VMs are running on the cluster and requesting the metrics.

This was first brought up on [CNV-38661](https://issues.redhat.com/browse/CNV-38661) and mitigated by #11484, but even with this mitigation in place, the root issue still persists: the server runs in the virt-handler outside the VM's cgroup.

A proper solution will be to move back the downward metrics server from the virt-handler to the virt-launcher, so it will run in the same cgroup as the VM

**Comment by jcanocan** [comment]
@xpivarc Would you agree with this approach? Any objections?

**Comment by jcanocan** [comment]
This is tracked in https://issues.redhat.com/browse/CNV-41208

**Comment by xpivarc** [comment]
Well the #vms and #requests is bound right now so I wouldn't agree that the downward metrics server can lead to OOM or resource exhaustion. That being said we can explore if moving to the launcher is feasible

**Comment by vladikr** [comment]
I don't object as well. The only concern is whether virt-launcher will be able to see all the host related metrics.

**Comment by germag** [comment]
> I don't object as well. The only concern is whether virt-launcher will be able to see all the host related metrics.

Yes, that was my first approach. The majority of the metrics comes from libvirtd and the rest from world-readable files in `/proc`

**Comment by vladikr** [comment]
> > I don't object as well. The only concern is whether virt-launcher will be able to see all the host related metrics.
> 
> Yes, being part of the launcher was my first approach. The majority of the metrics comes from libvirtd and the rest from world-readable files in `/proc`

In some cases `/proc` would only contain files that are relevant for the container and will not include all the data that you would normally be able to retrive from the host side. 
That is my only concern. 
However, if you have verified that part, I'm +1 for moving this functionality to virt-launcher.
